decoder.go

```diff
@@ -16,9 +16,60 @@ const (
 	defaultMaxSize = 16000
 )
 
+type option[T Decoder | Encoder] func(d *T)
+
 // NewDecoder returns a new Decoder.
-func NewDecoder() *Decoder {
-	return &Decoder{cache: newCache(), maxSize: defaultMaxSize}
+func NewDecoder(opts ...option[Decoder]) *Decoder {
+	d := &Decoder{cache: newCache(), maxSize: defaultMaxSize}
+	for _, opt := range opts {
+		opt(d)
+	}
+	return d
+}
+
+// WithAliasTagDecoderOpt changes the tag used to locate custom field aliases.
+// The default tag is "schema".
+func WithAliasTagDecoderOpt(tag string) option[Decoder] {
+	return func(d *Decoder) {
+		d.SetAliasTag(tag)
+	}
+}
+
+// WithZeroEmptyDecoderOpt controls the behaviour when the decoder encounters empty values.
+// in a map.
+// If z is true and a key in the map has the empty string as a value
+// then the corresponding struct field is set to the zero value.
+// If z is false then empty strings are ignored.
+//
+// The default value is false, that is empty values do not change
+// the value of the struct field.
+func WithZeroEmptyDecoderOpt(z bool) option[Decoder] {
```

Comment:
Maybe this and `WithIgnoreUnknownKeysDecoderOpt()` shouldn't have any arguments, but just invert whatever the default is? Just a thought.